### ISSUE TITLE
Rule "vue/valid-template-root" should be disabled for slot templates

### ISSUE THREAD

**Issue by IvoPereira** [vuejs/eslint-plugin-vue#621]
**Tell us about your environment**

* **ESLint Version:** 4.19.1
* **eslint-plugin-vue Version:** 4.7.1
* **Node Version:** v10.6.0

**Please show your full configuration:**

```js
module.exports = {
  root: true,
  env: {
    node: true
  },
  'extends': [
    'plugin:vue/essential',
    "eslint:recommended",
    '@vue/airbnb'
  ],
  rules: {
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-unused-vars': [2, {'args': 'after-used', 'argsIgnorePattern': '^_'}],
    'no-param-reassign': [2, {'props': false}]
  },
  parserOptions: {
    parser: 'babel-eslint',
    "ecmaVersion": 8,
    "ecmaFeatures": {
      "impliedStrict": true,
      "classes": true
    }
  },
}
```

**What did you do? Please include the actual source code causing the issue.**

```vue
<template>
  <v-select
    class="elevation-0"
    label="Not being sorted"
    :items="employees"
    v-model="buy.sorter_id"
    item-text="full_name"
    item-value="id"
    dense
    solo
    clearable
    @input="handleSelectInput"
  >
    <template slot="selection" slot-scope="data">
      <v-flex xs2>
        <v-avatar size="25px">
          <img :src="data.item.safe_avatar_url"/>
        </v-avatar>
      </v-flex>
      <v-flex class='ml-1'>
        {{ data.item.full_name }}
      </v-flex>
    </template>
    <template slot="item" slot-scope="data">
      <v-list-tile-avatar>
        <img :src="data.item.safe_avatar_url" />
      </v-list-tile-avatar>
      <v-list-tile-content>
        <v-list-tile-title v-html="data.item.full_name"></v-list-tile-title>
      </v-list-tile-content>
    </template>
  </v-select>
</template>
```

**What did you expect to happen?**

I expected the code to run without warnings.

**What actually happened? Please include the actual, raw output from ESLint.**

> [eslint-plugin-vue]
> [vue/valid-template-root]
> The template root requires exactly one element.

This refers to the [valid-template-root](https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/valid-template-root.md) on the inner slot template snippet:

```vue
    <template slot="item" slot-scope="data">
      <v-list-tile-avatar>
        <img :src="data.item.safe_avatar_url" />
      </v-list-tile-avatar>
      <v-list-tile-content>
        <v-list-tile-title v-html="data.item.full_name"></v-list-tile-title>
      </v-list-tile-content>
    </template>
```

The rule shouldn't catch a slot template.

**Comment by armano2** [comment]
Issue is no longer present in v5.0.0-beta.3
[https://mysticatea.github.io/vue-eslint-demo/#....](https://mysticatea.github.io/vue-eslint-demo/#eJyNVstu4zYU/RVBm9mEcRugGyMNuu2mm+nSQEBJVzJnKFLlw7EmyL/3kLQkSnIwswgQnnPfL/m9rHVD5bF8dtQPkjt6OamieL4wS5JqFx5FUUtu7Z+nEtCFO6EV++1UJkryiiSof7QrKhKqK6w2jpqJPwoYjrowr0ciOzEX1sN10K38+Bi1zKuYFYMec3R1EGi9lK+K97QiL1x6ArvoNKQspX+tlnoKnrjhlbwRfwk1+GD0zFUj6WtM8++AJSsvSWyuR2GlDuKpHsj9VEaI2VoPwX3DHT+VN7VYulbStbjapxmLKEfpuCms+BG0nv4YrplWlBF9VxytqW9GH0OWj5a39Jp0X72Rp/KQmz1Mdhf/hxTALqDUxC+9ZL9/yUy8vxeLs7nQxcfHJwafD6tJuVOpYOnnRZLCOuaEpG0Cv1yI4rBK+lODK2e1Vo6UW7dmoZ1wkjCaZ9fLlfdsBF/W3qLKJ5GsvW1KF0TTVOGZceVDabwkWx7fsZ3KOuNrpw2zfiBTHp8eyhavRpg0jxHpyGF9mCHnTUKUZjW3xBpC403cWjsTYRQY/kS3COt+4IaYoo79IKMXXDU7UYX1ovzp7og4rlxUF3OYiXFGS4Ta0XUCG6p8192yi29JjrDhC4LkGTedXQEpkZ76isya+U5jDkhRo7SxIhOKcruR1WcUpw61i7bW5MBDVdUKnB/XTco4VYazSqMyXAVPbs1Y6sWEtFxKdzbad+cZ8qreWOykrrjcgEIpBHuvq0LhIoomVXaYUYO3hzB7O2OS7cDruQS9uFLDIoRxQJ8dr2Zzit6YHftKywnR1TfkJeUsomvHZ9ZQCooWAHHcrE8Y5r3dJAQBY0NvDV9a5s7Csoow6JTNPQivGmqXBxLFFiCJ3ktsnFC0cIY4ulvJDAo3hLVCIYlxg2LueT6nXnkLu/H7Zjcg5nKBwgpjCMnWfEiuUCcExewZ37TwlYmgof88VpaNgmRCoMmEVTy5TL1z40A65XfxdMBS9rgf07JfaGa+2WtwbZdYAop4uB0xSELFhfahNIPBDTZO0EoOe9h6iw82w3nHPbkw0eb8eosW8LZJ2AwjKthf8aGVwSQZo01OGLJkLghma9Fi/wCHYLWKqeLkrgREQ4zaFvnbX0hruqLB0Rq/4ptAHIOCw1SfKSdvbZ2DsHfIbZ1D90LN2Jtw51TDmZ56vWQl7I4M8cd+s+zI76QMYeBXhz1nU+vyVGcmjhN2hWMxoq9MJNi6V8vFzG0ap2Iard2Gu7BK3EY7B2up+fcdihWifL5yfAfmlZyw8D3egXcMxp+UO1SrO1C99zyYPYYtftuBYZoAfjyUqAH6+xU/6qJQev47hktQplv7UMYDh5xKXF9DVH78D02e1/A=)

or... maybe i don't understand something
